File: src/main/java/com/refinedmods/refinedstorage/item/ProcessorItem.java
package com.refinedmods.refinedstorage.item;

import net.minecraft.world.item.Item;

public class ProcessorItem extends Item {
    public ProcessorItem() {
        super(new Item.Properties());
    }

    public enum Type {
        RAW_BASIC("raw_basic"),
        RAW_IMPROVED("raw_improved"),
        RAW_ADVANCED("raw_advanced"),
        BASIC("basic"),
        IMPROVED("improved"),
        ADVANCED("advanced");

        final String name;

        Type(String name) {
            this.name = name;
        }

        public String getName() {
            return name;
        }
    }
}

File: src/main/kotlin/com/refinedmods/refinedstorage/item/ProcessorItem.kt
package com.refinedmods.refinedstorage.item

import com.refinedmods.refinedstorage.RS
import net.minecraft.item.Item

class ProcessorItem(type: Type) : Item(Properties().group(RS.MAIN_GROUP)) {
    enum class Type(override val name: String) {
        RAW_BASIC("raw_basic"), RAW_IMPROVED("raw_improved"), RAW_ADVANCED("raw_advanced"), BASIC("basic"), IMPROVED("improved"), ADVANCED("advanced");
    }

    init {
        this.setRegistryName(RS.ID, type.name + "_processor")
    }
}